sdk/storage/storage-blob/src/Clients.ts

```diff
@@ -924,6 +925,9 @@ export class BlobClient extends StorageClient {
     options?: StoragePipelineOptions
   ) {
     options = options || {};
+    if (!options.httpClient) {
+      options.httpClient = new DefaultHttpClient();
```

Comment:
Since we do this in every client, should we add a comment that it is done in order to share HttpClients? 

Also, if we really want to share/reuse, should we have the library itself create and cache a default HttpClient instead of just ones that are inheriting another's pipeline?

Reply:
Not in leaf clients. Now only in clients that can be parents. I thought about caching a single instance, but was afraid because it is one of the two hardest problem in computer science!

Reply:
Well, since we don't have to worry about cross-package dependencies like `core-tracing` I think it would be enough to have a module that did `export const DefaultClient = new DefaultHttpClient()` and import that to use if `options.httpClient` is empty

Reply:
I just did almost that, except with a `getCachedDefaultHttpClient()` wrapper around the const.